Whiskey/en-US/about_Whiskey_Writing_Tasks.help.txt

```diff
@@ -0,0 +1,145 @@
+
+    # Tools
+
+    Whiskey can automatically install some tools for your task. The path to that tool is passed to your task in the `TaskParameter` parameter. 
+
+    To tell Whiskey what tools you depend on, add a `Whiskey.RequiresTool` attribute. It has two parameters: the name of the tool, and the name of the key to use in the `TaskParameter` hashtable where the path to your tool is saved.
+
+    For example, given this task:
+
+        function FUNCTION_NAME
+        {
+            [Whiskey.Task(''TASK_NAME'')]
+            # The second parameter is the name of the key in the $TaskParamter hashtable where Whiskey should put the path to the Node executable it installs.
+            [Whiskey.RequiresTool(''Node'',''NodePath'')]
+            [CmdletBinding()]
+            param(
+                $TaskContext,
+
+                [hashtable]
+                $TaskParameter
+            )
+
+            $nodePath = Assert-WhiskeyNodePath -Path $TaskParameter[''NodePath''] -ErrorAction Stop
+        }
+
+    Whiskey will install the latest version of Node, and put the path to the Node executable in the `NodePath` key in the `TaskParameter` hasthable. You can see that the task is using `Assert-WhiskeyPath` to ensure the `NodePath` parameter points to an existing path.
+
+    ## Installing a Specific Tool Version
+
+    If your task requires a specific version of a tool, use the `Whiskey.RequiresTool` attribute's `Version` property:
+
+        [Whiskey.RequiresTool('NodeModule::nsp', 'NspPath', Version='2.7.0')]
+
+    If you want the user to be able to control what version of a tool to use, the `Whiskey.RequiresTool` attribute has a `VersionParameterName` property that should be set to a property name users can use in their whiskey.yml file to control what version of a tool to use. The default name is `Version`. For example, if your `Whiskey.RequiresTool` attribute looks like this:
+
+         [Whiskey.RequiresTool('NodeModule::nsp', 'NspPath', VersionParameterName='NspVersion')]
+
+    A user could use the `NspVersion` property to control what version of Node they want to use:
+
+         BuildTasks:
+         - TASK_NAME:
+            NspVersion: 8.9.4
+
+    In the example above, Whiskey would install version 8.9.4 of ths NSP node module.
+
+    Whiskey supports installing:
+
+    * Node
+    * Node Modules
+
+    ## Node
+
+    To install Node, add this to your task:
+
+         [Whiskey.RequiresTool('Node','NodePath')]
+
+    The `NodePath` string is the name of the key where the path to Node should be saved in the `TaskParameter` hashtable when calling your task.
+
+    To control what version of Node to use, users must specify an `engines.node` property in their package.json file, e.g.
+
+       {
+            "engines": {
+                "node": "^8.9.4"
+            }
+       }
+
+    In other words, Whiskey ignores your task's `Whiskey.RequiresTool` attribute's VersionParameterName property.

```

Comment:
This sentence seems confusing. Doesn't the version defined in the whiskey.yml override the version in package.json? If so, then it's no so much that `VersionParameterName` property was ignored but that it wasn't defined, so the fallback is version from package.json. Actually should we include a quick sentence or list that describes the order of precedence when finding what version to install? e.g. whiskey.yml -> package.json -> latest LTS